pandas/core/internals.py

```diff
@@ -1807,16 +1807,20 @@ def _try_operate(self, values):
         return values.view('i8')
 
     def _try_coerce_args(self, values, other):
-        """ provide coercion to our input arguments
-            we are going to compare vs i8, so coerce to integer
-            values is always ndarra like, other may not be """
+        """ Coerce values and other to dtype 'i8'. NaN and NaT convert to
+            the smallest i8, and will correctly round-trip to NaT if converted
+            back in _try_coerce_result. values is always ndarray-like, other
+            may not be """
         values = values.view('i8')
+
```

Comment:
I'm sorry, I still don't see the common functionality. You're talking about this function:

```python
def _try_coerce_args(self, values, other):
    """ Coerce values and other to dtype 'i8'. NaN and NaT convert to
        the smallest i8, and will correctly round-trip to NaT if converted
        back in _try_coerce_result. values is always ndarray-like, other
        may not be """
    values = values.view('i8')

    if is_null_datelike_scalar(other):
        other = tslib.iNaT
    elif isinstance(other, datetime):
        other = lib.Timestamp(other).asm8.view('i8')
    elif hasattr(other, 'dtype') and com.is_integer_dtype(other):
        other = other.view('i8')
    else:
        other = np.array(other, dtype='i8')

    return values, other
```
and this one?
```python
def _infer_fill_value(val):
    """
    infer the fill value for the nan/NaT from the provided scalar/ndarray/list-like
    if we are a NaT, return the correct dtyped element to provide proper block construction

    """

    if not is_list_like(val):
        val = [val]
    val = np.array(val,copy=False)
    if is_datetimelike(val):
        return np.array('NaT',dtype=val.dtype)
    elif is_object_dtype(val.dtype):
        dtype = lib.infer_dtype(_ensure_object(val))
        if dtype in ['datetime','datetime64']:
            return np.array('NaT',dtype=_NS_DTYPE)
        elif dtype in ['timedelta','timedelta64']:
            return np.array('NaT',dtype=_TD_DTYPE)
    return np.nan
```

I haven't seen any code block that looks similar to ``_try_coerce_args`` elsewhere either. Could you give an example?